src/backend/aspen/database/models/sample.py

```diff
@@ -46,6 +48,29 @@ class RegionType(enum.Enum):
 )
 
 
+def create_public_id(context) -> str:
+    current_parameters: Mapping[
```

Comment:
Could we just make this a helper method that just takes in a `group_id`?

Reply:
so i looked into that, the issue is that if you don't have the context passed in you can't create a session from the context engine and then we couldn't query to find the group prefix :(